### PR TITLE
AssertNoError migration

### PR DESCRIPTION
**Pull request by aq17**
Replace `contract.AssertNoError` with `contract.AssertNoErrorf` to fix lint failures, as `AssertNoError` is now deprecated

### CODE REVIEW

File: upgrade/steps.go [hunk 1]
```diff
@@ -841,24 +834,19 @@ func AddAutoAliasing(ctx Context, repo ProviderRepo, providerName string) (step.
...
+	migrationSteps, err := migrationSteps(ctx, repo, providerName, "Add AutoAliasing", AutoAliasingMigration)
+	if err != nil {
+		return nil, err
 	}
-	return step.Combined("Replace AssertNoError", steps...), nil
+	steps = append(steps, migrationSteps)
+	return step.Combined("Add AutoAliasing", steps...), nil
+}
+
+func ReplaceAssertNoError(ctx Context, repo ProviderRepo, providerName string) (step.Step, error) {
```
Comment (aq17) [inline upgrade/steps.go:849]: `providerName` is unused here, but allows `migrationSteps()` to take in `migrationFunc func(resourcesFilePath, providerName string) (bool, error)` as an arg

File: upgrade/migrations.go [hunk 2]
```diff
@@ -167,3 +167,58 @@ func AutoAliasingMigration(resourcesFilePath, providerName string) (bool, error)
 	}
 	return true, nil
 }
+
+func AssertNoErrorMigration(resourcesFilePath, providerName string) (bool, error) {
+	return replaceAstFunction(resourcesFilePath,
```
Comment (guineveresaenger) [inline upgrade/migrations.go:172]: What does AST mean here?

**Review by guineveresaenger — COMMENTED** [review]
Is there an issue or design doc associated with this change?

File: upgrade/migrations.go (hunk 2 above)
Comment (iwahbe) [inline upgrade/migrations.go:172]: This is super easy to read. Thank you!

File: upgrade/steps.go (hunk 1 above)
Comment (iwahbe) [reply upgrade/steps.go:849]: It looks like you pass `provideName` to `migrationSteps` here.

Either way, this kind of clarifying comment is great as a comment in your code.

**Review by iwahbe — APPROVED** [review]
I have 1 nit, but otherwise LGTM!

**Pull request merged**